### ISSUE TITLE
Property 'toast' does not exist on type 'JQueryStatic'

### ISSUE THREAD

**Issue by sagg629** [Script47/Toast#5]
Hi! Happy to found this mini library to manage Bootstrap toasts easily.

I'm working on a Bootstrap-Angular-Typescript project and I'm getting this error on NPM console each time I compile the project:
> `Property 'toast' does not exist on type 'JQueryStatic'`

I've tried using the npm's `bs4-toast` dependency (as stated in #3, my preferred BTW) and importing manually the `toast.min.js` file, all of them with the same error.
Also followed [these](https://stackoverflow.com/a/32052431/3135458) suggestions, without advance.

Can you revise this?
Also would be great to include the `bs4-toast` dependency into the master's README.

Thanks in advance.

**Comment by Script47** [comment]
@sagg629,

I've never come across this issue before however I did find this with a quick search: https://stackoverflow.com/questions/35071120/property-valhooks-does-not-exist-on-type-jquerystatic

Does that help?

It looks like a TypeScript issue rather than one related to the library itself so I'm going to close this, feel free to let me know if it is otherwise.

**Comment by sagg629** [comment]
Hi @Script47, thanks for your fast response.

I've checked your link but I got the real deal from [here!](https://stackoverflow.com/questions/32050645/how-to-use-jquery-with-typescript).
So, after some hours of try-and-failure I've solved my issue by modifying some configuration parameters that I want to share here just for future references:

1. Import `bs4-toast` and `@types/jquery` using `npm install bs4-toast @types/jquery --save`, making sure current `jQuery` version (if available) matches its `@types` counterpart.
```
"dependencies": {
    "@types/jquery": "^3.3.29",
    "bs4-toast": "^0.6.0",
    "jquery": "^3.3.29",
}
```

2. Enable `allowSyntheticDefaultImports` flag in `tsconfig.json` file (according to [TypeScript docs](https://www.typescriptlang.org/docs/handbook/compiler-options.html), to allow default imports from modules with no default export).
```
{
...
  "compilerOptions": {
     ...
     "esModuleInterop": true
  },
  "allowSyntheticDefaultImports": true
}
```

3. Add a new `"jquery"` entry into `types` property on `tsconfig.app.json` file.
```
"types": ["jquery"]
```

4. Reference library files into `angular.json` file.
```
"styles": [
    "src/styles.css",
    "node_modules/bootstrap/dist/css/bootstrap.min.css",
    ...
    "node_modules/bs4-toast/dist/toast.min.css"
],
"scripts": [
    "node_modules/jquery/dist/jquery.min.js",
    "node_modules/bootstrap/dist/js/bootstrap.min.js",
    ...
    "node_modules/bs4-toast/dist/toast.min.js"
]
```

5. In the component from which toast will be called, add this below `import`s
```
declare var $: any;
```

Hope this helps.
:heart:
